scripts/hal_alignment/maf_to_fasta.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python3
+
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Convert each block of a MAF alignment to a FASTA file.
+
+Examples::
+    python maf_to_fasta.py --genomes-file genomes.txt input.maf output_dir/
+
+"""
+
+from argparse import ArgumentParser
+import json
+import os
+from pathlib import Path
+import re
+from typing import Dict, Iterable, Pattern, Tuple, Union
+
+from Bio.AlignIO.MafIO import MafIterator
+from Bio import SeqIO
+from Bio.SeqRecord import SeqRecord
+
+
+def compile_maf_src_regex(genome_names: Iterable[str]) -> Pattern[str]:
+    """Make a MAF src field regex for the given genome names.
+
+    In the UCSC multiple alignment format (MAF), the src field can be of the form '<genome>.<seqid>',
```

Comment:
```suggestion
    In the UCSC MAF file, the src field can be of the form '<genome>.<seqid>',
```
(if you like my previous suggestion)